From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MeFisto94 <MeFisto94@users.noreply.github.com>
Date: Mon, 16 Aug 2021 03:03:13 +0200
Subject: [PATCH] Allow Cheats to be used by default (ctrl+shift+F1...F11)


diff --git a/Cheat.cs b/Cheat.cs
index e7213abda77682f75ba36e13f16d7600308d0251..80d663431d68f37757b0334244bf23035c8a7c13 100644
--- a/Cheat.cs
+++ b/Cheat.cs
@@ -239,7 +239,9 @@ public class Cheat : MonoBehaviour_ID_Network
 		{
 			return false;
 		}
-		return AllowCheatsForUser(player.steamID);
+		// Sharkkit Start - Allow Cheats by default
+		return true; // return AllowCheatsForUser(player.steamID);
+		// Sharkkit End
 	}
 
 	public static bool AllowCheatsForUser(CSteamID cSteamID)
