From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MeFisto94 <MeFisto94@users.noreply.github.com>
Date: Mon, 16 Aug 2021 14:52:02 +0200
Subject: [PATCH] First steps at implementing the chat for IO


diff --git a/ChatManager.cs b/ChatManager.cs
index 0e4e4b8fdf204aa0990b27f0eb0005bd3a542de1..56b6d505999e9ac59eaff24c24e7ba605923a126 100644
--- a/ChatManager.cs
+++ b/ChatManager.cs
@@ -1,7 +1,7 @@
 using Steamworks;
 using UnityEngine;
 
-public class ChatManager : MonoBehaviour_Network
+public class ChatManager : MonoBehaviour_Network, SharkkitAPI.IChatManager
 {
 	public delegate void OnRecieveChatMessageAction();
 
@@ -12,6 +12,11 @@ public class ChatManager : MonoBehaviour_Network
 
 	public static event OnRecieveChatMessageAction OnRecieveChatMessage;
 
+	void SharkkitAPI.IChatManager.LocalDebugChatMessage(string message)
+	{
+		LocalDebugChatMessage(message);
+	}
+
 	private void Awake()
 	{
 		ComponentManager<ChatManager>.Value = this;
@@ -65,6 +70,11 @@ public class ChatManager : MonoBehaviour_Network
 		chatManager.SendChatMessage(message, network_Player.steamID);
 	}
 
+	void SharkkitAPI.IChatManager.SendDebugChatMessage(string message)
+	{
+		SendDebugChatMessage(message);
+	}
+
 	public static void LocalDebugChatMessage(string message)
 	{
 		ChatManager chatManager = null;
diff --git a/GameManager.cs b/GameManager.cs
index 112b32d0b92f9c78f0217ba185e3625923d574c2..41eae3883558a9245d2b33635b13600848799c19 100644
--- a/GameManager.cs
+++ b/GameManager.cs
@@ -92,6 +92,11 @@ public class GameManager : SingletonGeneric<GameManager>, SceneEventInterface, S
 
 	string SharkkitAPI.GameManager.ApiImplVersion => Settings.ImplVersionNumberText;
 
+	public SharkkitAPI.IChatManager GetChatManager()
+	{
+		return ComponentManager<ChatManager>.Value;
+	}
+
 	public SharkkitAPI.GameManager Instance => Singleton;
 	// Sharkkit End
 
